runtime/src/bank.rs

```diff
@@ -2074,7 +2090,10 @@ impl Bank {
         );
 
         // Add a bogus executable native account, which will be loaded and ignored.
-        let account = native_loader::create_loadable_account(name);
+        let account = native_loader::create_loadable_account(
+            name,
+            self.inherit_specially_retained_account_balance(&existing_genuine_program),
```

Comment:
this new inherit_specially_retained_account_balance call is justified by this: (test_add_native_program_inherited_cap_while_replacing)

Reply:
bug description (re-post from https://github.com/solana-labs/solana/pull/13403#discussion_r535955941):

 > Also, when replacing, we must consider the balance of existing program account otherwise we would cause capitalization mismatch if the replaced native program isn't 1 lamports.

Reply:
Also, I later noticed that we can meaninglessly send to sysvars, but not to executable accounts. So, this can't happen on the wild. But, addressing and adding tests should never hurt us not much. :)